src/utils/staking/index.ts

```diff
@@ -43,3 +63,133 @@ export const buildStakingOutput = (
   }
   return psbtOutputs;
 };
+
+/**
+ * Get the staking term, it will ignore the `stakingTimeBlocks` and
+ * use the value from params if the min and max staking time blocks are the same
+ *
+ * @param {Params} params - The staking parameters.
+ * @param {number} term - The staking term.
+ * @returns {number} - The staking term.
+ */
+export const getStakingTerm = (params: Params, term: number) => {
+  // check if term is fixed
+  let termWithFixed;
+  if (params.minStakingTimeBlocks === params.maxStakingTimeBlocks) {
+    // if term is fixed, use the API value
+    termWithFixed = params.minStakingTimeBlocks;
+  } else {
+    // if term is not fixed, use the term from the input
+    termWithFixed = term;
+  }
+  return termWithFixed;
+};
+
+/**
+ * Validate the staking transaction input data.
+ *
+ * @param {number} stakingAmountSat - The staking amount in satoshis.
+ * @param {number} stakingTerm - The staking term in blocks.
+ * @param {Params} params - The staking parameters.
+ * @param {UTXO[]} inputUTXOs - The input UTXOs.
+ * @param {number} feeRate - The Bitcoin fee rate in sat/vbyte
+ * @throws {StakingError} - If the input data is invalid.
+ */
+export const validateStakingTxInputData = (
+  stakingAmountSat: number,
+  stakingTerm: number,
+  params: Params,
+  inputUTXOs: UTXO[],
+  feeRate: number,
+) => {
+  if (
+    stakingAmountSat < params.minStakingAmountSat ||
+    stakingAmountSat > params.maxStakingAmountSat
+  ) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid staking amount",
+    );
+  }
+
+  if (
+    stakingTerm < params.minStakingTimeBlocks ||
+    stakingTerm > params.maxStakingTimeBlocks
+  ) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid staking term",
+    );
+  }
+
+  if (inputUTXOs.length == 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "No input UTXOs provided",
+    );
+  } else if (feeRate <= 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid fee rate",
+    );
+  }
+}
+
+/**
+ * Validate the parameters for staking.
+ * 
+ * @param {Params} params - The staking parameters.
+ * @throws {StakingError} - If the parameters are invalid.
+ */
+export const validateParams = (params: Params) => {
```

Comment:
Have we double checked that we apply the same verifications for the params as the golang library?